okhttp/src/main/java/okhttp3/internal/http2/Http2Connection.kt

```diff
@@ -691,21 +693,38 @@ class Http2Connection internal constructor(builder: Builder) : Closeable {
      * order than received.
      */
     fun applyAndAckSettings(clearPrevious: Boolean, settings: Settings) {
-      var delta = 0L
-      var streamsToNotify: Array<Http2Stream>? = null
+      var delta: Long
+      var streamsToNotify: Array<Http2Stream>?
+      var newPeerSettings: Settings
       synchronized(writer) {
         synchronized(this@Http2Connection) {
-          val priorWriteWindowSize = peerSettings.initialWindowSize
-          if (clearPrevious) peerSettings.clear()
-          peerSettings.merge(settings)
-          val peerInitialWindowSize = peerSettings.initialWindowSize
-          if (peerInitialWindowSize != -1 && peerInitialWindowSize != priorWriteWindowSize) {
-            delta = (peerInitialWindowSize - priorWriteWindowSize).toLong()
-            streamsToNotify = if (streams.isNotEmpty()) streams.values.toTypedArray() else null
+          val previousPeerSettings = peerSettings
+          if (clearPrevious) {
+            newPeerSettings = settings
+          } else {
+            newPeerSettings = Settings()
```

Comment:
Maybe cleaner as an apply with merge inside the block?

Reply:
Good idea. Can do.